Makefile

```diff
@@ -64,18 +67,33 @@ busted: dependencies $(ROVER) ## Test Lua.
 nginx:
 	@ ($(NGINX) -V 2>&1) > /dev/null
 
-cpan:
+$(CPANM):
 ifeq ($(CPANM),)
 	$(error Missing cpanminus. Install it by running `curl -L https://cpanmin.us | perl - App::cpanminus`)
 endif
-	$(CPANM) --notest --installdeps ./gateway
+
+local/bin/carton: $(CPANM)
+	$(CPANM) --local-lib ./local --notest Carton
+
+cpan: $(CPANM)
+	$(CPANM) --local-lib ./local --notest --installdeps ./gateway
+
+PERL5LIB:=$(PWD)/local/lib/perl5:$(PERL5LIB)
+export PERL5LIB
+
+carton: export PERL_CARTON_CPANFILE=$(PWD)/gateway/cpanfile
+carton: export PERL_CARTON_PATH=$(PWD)/local
+carton: $(CARTON)
+carton:
+	$(CARTON) install --deployment --cached
+	$(CARTON) bundle 2> /dev/null
 
 find-file = $(shell find $(2) -type f -name $(1))
 
 prove: HARNESS ?= TAP::Harness
 prove: PROVE_FILES ?= $(filter-out $(call find-file, "*.t", examples/scaffold),$(call find-file, *.t, t examples))
 prove: export TEST_NGINX_RANDOMIZE=1
-prove: $(ROVER) nginx cpan ## Test nginx
+prove: $(ROVER) nginx ## Test nginx
```

Comment:
Why carton instead of cpan?

Reply:
Because it can produce .tar.gz of installed dependencies that can be used as a cache. 
Also it does a bit better locking as it has cpanfile.snapshot.

We removed it because it was pain to install it cross system, but now it should be ok with the latest trick to also install it into `./local`.